### PR TITLE
AssemblyInfo update fixes

### LINKED ISSUE #858: Update AssemblyInfo files failes for *.vb files
When using the GitVersion Task for VSTS the following error occurs on VB projects:

image

This is due to the c# assembly info being added..

Workaround: Disable the GitVersion task and build succeeds

### PR DESCRIPTION
**Pull request by asbjornu**
This PR makes the `AssemblyInfoFileUpdate` class support F# and Visual Basic as well as C# and with that fixes #858. It also adds a load of tests for F# and Visual Basic by making all of the `AssemblyInfoFileUpdate` tests into `[TestCase]` methods, and with that testing not just C#, but F# and VisualBasic as well.

### CODE REVIEW

**Comment by asbjornu** [comment]
Don't merge just yet. I discovered some bugs in a few of the tests.

**Comment by asbjornu** [comment]
There we go. Merge whenever! :smile:

File: src/GitVersionCore.Tests/JsonVersionBuilderTests.cs [hunk 1]
```diff
@@ -25,6 +25,6 @@ public void Json()
 
         var variables = VariableProvider.GetVariablesFor(semanticVersion, config, false);
         var json = JsonOutputFormatter.ToJson(variables);
-        json.ShouldMatchApproved();
+        json.ShouldMatchApproved(c => c.SubFolder("Approved"));
```
Comment (JakeGinnivan) [inline src/GitVersionCore.Tests/JsonVersionBuilderTests.cs]: I am actually really happy with the ShouldMatchApproved api :P

Comment (asbjornu) [inline src/GitVersionCore.Tests/JsonVersionBuilderTests.cs]: @JakeGinnivan It's beautiful! :smile:

**Comment by JakeGinnivan** [comment]
Sweet!

**Pull request merged**